src/nunit.analyzers.codefixes/UseAssertEnterMultipleScope/UseAssertEnterMultipleScopeCodeFix.cs

```diff
@@ -0,0 +1,99 @@
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using NUnit.Analyzers.Constants;
+
+namespace NUnit.Analyzers.UseAssertEnterMultipleScope
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    [Shared]
+    public class UseAssertEnterMultipleScopeCodeFix : CodeFixProvider
+    {
+        internal const string UseAssertEnterMultipleScopeMethod = "Use Assert.EnterMultipleScope method";
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(AnalyzerIdentifiers.UseAssertEnterMultipleScope);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (root is null)
+            {
+                return;
+            }
+
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            var node = root.FindNode(context.Span);
+
+            var expressionStatementSyntax = FindNearestParentOfType<ExpressionStatementSyntax>(node);
+            if (expressionStatementSyntax is null)
+            {
+                return;
+            }
+
+            var blockSyntax = FindBlockSyntax(node);
+            if (blockSyntax is null)
+            {
+                return;
+            }
+
+            var newSyntax = CreateUsingStatementSyntax(blockSyntax, expressionStatementSyntax.GetTrailingTrivia());
+
```

Comment:
If the node is an InvocationExpression of `Assert.Multiple` and  `ParenthesizedLambdaExpressionSyntax` has an `async` modifier and the return value of the method this statement is in is not `async Task` then you have to:
* Create a new MethodDeclarationSyntax replacing the return type with `async Task`
* Copy all statements, except replacing the original `expressionStatementSyntax` with `newSyntax`. See `UseAssertMultipleCodeFix` which also needs to copy statements.

You might think you can do two calls to `ReplaceNode` on for the MethodDeclaration and one for the statement, but I don't think that is allowed from what I remember.

You also need to check whether `System.Threading` is in the `using` list or use a full type name for  the return type.

Have fun.

Reply:
Thanks for the review Manfred. I opted for the solution with two `ReplaceNode` calls as I found this to easier (although I needed to find the new node in the updated document).